alot/commands/envelope.py

```diff
@@ -511,13 +514,16 @@ def apply(self, ui):
                     ui.notify(str(e), priority='error')
                     return
             else:
-                try:
-                    acc = settings.get_account_by_address(envelope['From'])
-                except NoMatchingAccount:
-                    envelope.sign = False
-                    ui.notify('Unable to find a matching account',
-                              priority='error')
-                    return
+                if envelope.account is None:
```

Comment:
In principle I agree and I'll try to find a better way, but if there was a way to guarantee that an envelope had an account attached to it, that whole dance wouldn't be necessary. However, enforcing that seems like an even more invasive change than what I did here so I stayed conservative at the cost of more error checking.

Reply:
PS: A thing I considered was making account_matching_address return None instead of raising NoMatchingAccount, this should make all code calling it less convoluted. I refrained from doing so since it was pretty invasive and I didn't want to start a side-discussion about error handling philosophies.